R/validate.R

```diff
@@ -0,0 +1,155 @@
+#' @include import-standalone-types-check.R
+validate_args <- function(model, times, control, cost, call = rlang::caller_env()) {
+  model <- rlang::arg_match(model, baguette_models, error_arg = "base_model",
+                            error_call = call)
+
+  if (!is.null(cost) & !(model %in% c("CART", "C5.0"))) {
+    cli::cli_abort("When using misclassification costs, {.arg base_model} should
+                    be either {.val CART} or {.val C5.0}.", call = call)
+  }
+  if (!is.matrix(cost)) {
+    check_number_decimal(cost, allow_null = TRUE, min = 0, call = call)
+  } else {
+    is_sq <- nrow(cost) == ncol(cost)
+    if (!is.numeric(cost) || !is_sq) {
+      cli::cli_abort("If {.arg cost} is a matrix, is must be numeric and square.",
+                     call = call)
+    }
+  }
+
+  check_number_whole(times, min = 2, call = call)
+
+  validate_control(control, call = call)
+
+  invisible(TRUE)
+
+}
+
+integer_B <- function(B) {
+  if (is.numeric(B) & !is.integer(B)) {
+    B <- as.integer(B)
+  }
+  B
+}
+
+# ------------------------------------------------------------------------------
+
+validate_y_type <- function(base_model, outcomes) {
+  hardhat::validate_outcomes_are_univariate(outcomes)
+
+  if (base_model == "C5.0") {
+    hardhat::validate_outcomes_are_factors(outcomes)
+  }
+
+}
+
+# ------------------------------------------------------------------------------
+
+model_failure <- function(x) {
+  if (inherits(x, "model_fit")) {
+    res <- inherits(x$fit, "try-error")
+  } else {
+    res <- inherits(x, "try-error")
+  }
+  res
+}
+
+check_for_disaster <- function(x, call = rlang::caller_env()) {
+  x <- dplyr::mutate(x, passed = !purrr::map_lgl(model, model_failure))
+
+  if (sum(x$passed) == 0) {
+    if (inherits(x$model[[1]], "try-error")) {
+      msg <- as.character(x$model[[1]])
+    } else {
+      if (inherits(x$model[[1]], "model_fit")) {
+        msg <- as.character(x$model[[1]]$fit)
+      } else msg <- NA
+    }
+
+    if (!is.na(msg)) {
+      # escape any brackets in the error message
```

Comment:
A long way to get there but this should make much much nicer error messages when models fail